docs/contributors/setup/set-up-git-for-ubuntu-desktop-work.md

```diff
@@ -0,0 +1,458 @@
+:::
+
+Let's browse the content of the `upstreamvcs` repository:
+
+```{terminal}
+:copy:
+:host:
+:dir: gnome-control-center
+:user:
+:input: git remote show upstreamvcs
+
+* remote upstreamvcs
+[…]
+  HEAD branch: main
+  Remote branches:
+[…]
+    gnome-49                                  tracked
+[…]
+```
+
+The branches within `origin/upstream` on Salsa, such as the `origin/upstream/latest` branch, tag unmodified releases from the GNOME `upstreamvcs` remote, without the Debian packaging and changes. As a result, the local `upstream/latest` branch tracks the `main` branch of the `upstreamvcs` repository.
+
+You can check out any of those branch locally. We can track any additional branches as needed for cherry-picking fixes.
+
+Our convention is to check out the `upstreamvcs/main` branch locally and call it `upstreamvcs-main`:
+
+```{terminal}
+:copy:
+:host:
+:dir: gnome-control-center
+:user:
+:input: git checkout -b upstreamvcs-main upstreamvcs/main
+
+Branch 'upstreamvcs-main' set up to track remote branch 'main' from 'upstreamvcs'.
+Switched to a new branch 'upstreamvcs-main'
+```
+
+
+## Optional configuration
+
+The following configuration can simplify certain tasks on Ubuntu Desktop projects.
+
+### Git command aliases
+
+Enable Git command aliases:
+
+```{code-block} ini
+:caption: `~/.config/git/config`
+
+[alias]
+    matching-tags = "!f() { git for-each-ref --sort=creatordate --format '%(refname)' refs/tags| grep \"$1\"| sed s,^refs/tags/,,; }; f"
+    last-tags = matching-tags '.*'
+    last-match-tag = "!f() { git matching-tags $1 | tail -n1; }; f"
+    last-tag = last-match-tag '.*'
+    last-debian-tag = last-match-tag debian/
+    last-ubuntu-tag = last-match-tag ubuntu/
+    ubuntu-delta = "!f() { git diff $(git last-debian-tag) -- ${1:-debian} ':(exclude)debian/changelog'; }; f"
+```
+
+### Exporting the build area
+
+Export the `../build-area/` directory before building with the `git-buildpackage` tool:
+
+```{code-block} ini
+:caption: `~/.gbp.conf`
+
+[buildpackage]
+export-dir = ../build-area/
+```
+
+For details, see the `gbp.conf(5)` and `gbp-buildpackage` man pages.
+
+### Merging the changelog automatically
+
+To merge the changelog automatically, enable the `dpkg-mergechangelogs` tool in the `~/.config/git/config` file:
+
+```{code-block} ini
+:caption: `~/.config/git/config`
+
+[merge "dpkg-mergechangelogs"]  
+    name = debian/changelog merge driver
+    driver = dpkg-mergechangelogs -m %O %A %B %A
+```
+
+Create the `~/.config/git/attributes` file and set `dpkg-mergechangelogs` as the default strategy in it:
+
+```{code-block} ini
+:caption: `~/.config/git/attributes`
+
+debian/changelog merge=dpkg-mergechangelogs
+```
+
+:::{warning}
+This automation might not work when Ubuntu packages use a higher epoch than the Debian ones. In that case, fix the changelog manually.
```

Comment:
We don't have `epoch` described in [version string format](https://documentation.ubuntu.com/project/how-ubuntu-is-made/concepts/version-strings/) nor in the Glossary. In the meantime, let's link to Debian:

```suggestion
This automation might not work when Ubuntu packages use a higher [epoch](https://www.debian.org/doc/debian-policy/ch-controlfields.html#version) than the Debian ones. In that case, fix the changelog manually.
```